### DISCUSSION TITLE
Is the script supposed to check in/display companions or only travelers under the same confirmation?

### DISCUSSION THREAD

**Discussion by himnim** [jdholtz/auto-southwest-check-in#126] [Q&A] [answered]
I had previously configured a reservation for my sister-in-law and it successfully checked in/displayed all 4 travelers' boarding positions. I don't remember if her husband was her companion at that time or if it was before they received their companion pass.

I configured a reservation for a friend and it only checked in/displayed his boarding position and not his companion, which is technically under a different confirmation but linked to his (by virtue of being a companion). It's somewhat of a moot point since companions are now configured to board consecutive with the main ticketholder on SW's backend, but I'm curious if this is intended before I submit a bug report.

**Comment by jdholtz** [comment]
This is not intended, but I did not realize this issue until now. I will look further into this to see if I can automatically grab the companion's information from the current reservation. For now, a workaround is to input the companion's reservation information separately as companions still need to check in separately

**Reply by himnim** [reply]
Yeah, and as I mentioned it's sort of moot as SW's backend puts them consecutively with the main reservation. Do you want me to submit a bug report for it or consider it internally tracked?

**Reply by jdholtz** [reply]
Go ahead. This is an important issue so it would be great for there to be a bug about this.

**Comment by jdholtz** [comment, marked as answer]
Hey @himnim, I looked into this a bit more. When you just enter a reservation (without logging in), the companion information is not available, meaning this script wouldn't be able do anything in regards to automatically scheduling a companion's check-in. If you log in, you can get the information, but it isn't easy. Additionally, this would be unreliable to extract in the script because it would not handle modifications to companion passes. 

Therefore, automatically checking in companions *could* possibly work only for accounts, but not at all for reservations (which I believe is your situation). I think the best approach to this is just to let people know to set up the script for a companion as well (whether that be their account or just a single reservation) in the README.

**Reply by himnim** [reply]
That makes sense. I'll skip the bug report and either manually check-in companions or enter their reservation/account as well. Thanks for looking into it!

**Comment by dpbraune** [comment]
For quite a while now (perhaps 3-4 months) manual check-in of trips with a companion check-in the companion with a consecutive boarding slot automatically.  Recently, I was C-3 and my companion automatically received C-2. I would assume auto-check-in would trigger the same response from the SWA site.

**Reply by jdholtz** [reply]
Yes, I believe Southwest does make sure the companion will get a boarding pass next to the main person. However, I think the companion still has to check in separately. For this script, you’ll need to type in the companion’s information for that to happen.
